src/components/form-component.js

```diff
@@ -33,256 +33,270 @@ const propTypes = {
+          return getField(formValue, field).errors;
+        }
 
-      if (!shallowEqual(fieldErrors, currentErrors)) {
-        validityChanged = true;
-      }
+        const fieldErrors = getValidity(errorValidator, nextValue);
 
-      return fieldErrors;
-    });
+        if (!shallowEqual(fieldErrors, currentErrors)) {
+          validityChanged = true;
+        }
 
-    const fieldsErrors = merge(
-      invertValidity(fieldsValidity),
-      fieldsErrorsValidity
-    );
+        return fieldErrors;
+      });
 
-    // Compute form-level validity
-    if (!fieldsValidity.hasOwnProperty('') && !fieldsErrorsValidity.hasOwnProperty('')) {
-      fieldsErrors[''] = false;
-    }
+      const fieldsErrors = merge(
+        invertValidity(fieldsValidity),
+        fieldsErrorsValidity
+      );
+
+      // Compute form-level validity
+      if (!fieldsValidity.hasOwnProperty('') && !fieldsErrorsValidity.hasOwnProperty('')) {
+        fieldsErrors[''] = false;
+      }
 
-    if (validityChanged) {
-      dispatch(actions.setFieldsErrors(model, fieldsErrors));
+      if (validityChanged) {
+        dispatch(s.actions.setFieldsErrors(model, fieldsErrors));
+      }
     }
-  }
 
-  handleValidSubmit() {
-    const {
-      dispatch,
-      model,
-      modelValue,
-      onSubmit = identity,
-    } = this.props;
+    handleValidSubmit() {
+      const {
+        dispatch,
+        model,
+        modelValue,
+        onSubmit = identity,
+      } = this.props;
 
-    dispatch(actions.setPending(model));
+      dispatch(s.actions.setPending(model));
 
-    return onSubmit(modelValue);
-  }
+      return onSubmit(modelValue);
+    }
 
-  handleInvalidSubmit() {
-    const {
-      dispatch,
-      model,
-    } = this.props;
+    handleInvalidSubmit() {
+      const {
+        dispatch,
+        model,
+      } = this.props;
 
-    dispatch(actions.setSubmitFailed(model));
-  }
+      dispatch(s.actions.setSubmitFailed(model));
+    }
 
-  handleReset(e) {
-    if (e) e.preventDefault();
+    handleReset(e) {
+      if (e) e.preventDefault();
 
-    const {
-      model,
-      dispatch,
-    } = this.props;
+      const {
+        model,
+        dispatch,
+      } = this.props;
 
-    dispatch(actions.reset(model));
-  }
+      dispatch(s.actions.reset(model));
+    }
 
-  handleSubmit(e) {
-    if (e) e.preventDefault();
+    handleSubmit(e) {
+      if (e) e.preventDefault();
 
-    const {
-      model,
-      modelValue,
-      formValue,
-      onSubmit,
-      dispatch,
-      validators,
-      errors: errorValidators,
-    } = this.props;
+      const {
+        model,
+        modelValue,
+        formValue,
+        onSubmit,
+        dispatch,
+        validators,
+        errors: errorValidators,
+      } = this.props;
 
-    const formValid = formValue
-      ? formValue.valid
-      : true;
+      const formValid = formValue
+        ? formValue.$form.valid
```

Comment:
Don't worry, this isn't a hidden bug - it is expected that the `model` of `<Form>` is always a `form`, which means it always has a `$form` prop.